src/moonlink/src/storage/wal.rs

```diff
@@ -196,6 +238,11 @@ impl WalManager {
         format!("wal_{file_number}.json")
     }
 
+    /// Get the file system accessor for testing purposes
+    pub fn get_file_system_accessor(&self) -> Arc<dyn BaseFileSystemAccess> {
```

Comment:
If it's only for testing purpose, I personally prefer provide another factory function which takes an accessor.
The benefit is less exposed interface:
- It's easy to call `get_file_system_accessor` in production when it shouldn't (you didn't even mark it as test-only?)
- It's impossible to access and expose if we use factory function